tests/testthat/test-050-makeApiCall.R

```diff
@@ -37,6 +41,38 @@ test_that(
   }
 )
 
+test_that(
+  "makeApiCall deals with curl timeouts",
+  {
+    h <- new_handle(timeout = 1L)
+    goodVersionPOST <- structure(
+      list(url = "https://redcap.vanderbilt.edu/api/",
+           status_code = 200L,
+           content = charToRaw("13.10.3"),
+           headers=structure(list(
+             'Content-Type'="text/csv; charset=utf-8"
+           )),
+           class = c("insensitive", "list")),
+      class = "response")
+    e <- structure(
+           list(message = "Timeout was reached: [redcap.vanderbilt.edu] Operation timed out after 300001 milliseconds with 0 bytes received", 
+                call = curl_fetch_memory("https://redcap.vanderbilt.edu/api/params", handle = h)
+                ),
+           class = c("simpleError", "error", "condition")
+    )
+    
+    x <- 1
+    stub(makeApiCall, "httr::POST", function(...)
+      if(x==1) { x <<- 2; stop(e) } else {goodVersionPOST})
+    
+    response <- makeApiCall(rcon, 
```

Comment:
I feel like this isn't quite the full test we want. Since `rcon` is utilizing the retry strategy, and the stub only throws the error when x == 1. After the retries, `response` ends up being assigned `goodVersionPOST`, which doesn't allow us to process the error message......Oh, I just got it.  This test shows that makeApiCall recovers from the socket error gracefully.

I still think we would benefit from one more test.  Something along the lines of

```
rcon_temp <- rcon
rcon_temp$set_retries(1)
x <- 1
response <- makeApiCall(rcon, 
                        body = list(content = "version", 
                                    format = "csv"))
expect_error(redcapError(response, "A network error has occurred")). 
```

This test would demonstrate that we are sending the desired error message to the user if the socket error persists over all of the retries.

Reply:
Will do.